challenge.rb

```diff
@@ -1,40 +1,43 @@
 
 def capitalize_each_string(input)
-  #implement your solution here
-end
-
-def fetch_the_dog(input)
-  #implement your solution here
-end
-
-def no_dogs_allowed(input)
-  #implement your solution here
-end
-
-def count_the_animals(input)
-  #implement your solution here
-end
-
-def fetch_the_first_two(input)
-  #implement your solution here
-end
-
-def fetch_CD_animals(input)
-  #implement your solution here
-end
-
-## DO NOT CHANGE CODE BELOW THIS LINE ##
-
-animals = ["cat", "moose", "dog", "bird"]
-
-p capitalize_each_string(animals) == ["Cat", "Moose", "Dog", "Bird"]
-
-p fetch_the_dog(animals) == ["dog"] 
-
-p no_dogs_allowed(animals) == ["cat", "moose", "bird"]
-
-p count_the_animals(animals) == 4
-
-p fetch_the_first_two(animals)  == ["cat", "moose"]
-
-p fetch_CD_animals(animals) == ["cat", "dog"]
+    input.map { |item| item.capitalize }
+  end
+  
+  def fetch_the_dog(input)
+    input.find_all { |animal| animal == "dog" }
+  end
+  
+  def no_dogs_allowed(input)
+    input = input.dup
+    input.delete("dog")
+    return input
+  end
+  
+  def count_the_animals(input)
+    input.length
```

Comment:
length isn't actually an enumerable but there's good reasons for familiarizing yourself with it and it's counterparts (eg. `.size`, `.count`) use cases. 

see this article: http://batsov.com/articles/2014/02/17/the-elements-of-style-in-ruby-number-13-length-vs-size-vs-count/

https://ruby-doc.org/core-2.5.1/Enumerable.html#method-i-count